src/es-module-shims.js

```diff
@@ -128,7 +128,7 @@ async function resolveDeps (load, seen) {
       }
       // dynamic import
       else {
-        resolvedSource += source.slice(lastIndex, start) + 'importShim' + source.slice(start + 6, end) + JSON.stringify(load.r) + ', ';
+        resolvedSource += source.slice(lastIndex, dynamicImportIndex) + 'importShim(' + JSON.stringify(load.r) + ', ' + source.slice(start, end);
```

Comment:
So this was originally done this way because the first version of the lexer didn't have the ability to match the opening and closing brackets around the `import( )` expression, so instead if `importShim(idExpr, parentURL)`, it was converted into `importShim(parentURL, idExpr` so that it could work out.

Now that we do support this, it could be worth switching the `importShim` function back around in the definition above (`async function importShim (id` line), and then simplifying this line to:

```suggestion
        resolvedSource += source.slice(lastIndex, dynamicImportIndex + 6) + 'Shim(' + source.slice(start, end) + ', ' + JSON.stringify(load.r);
```

Reply:
You meant swapping the `arguments` check inside `async function importShim`, right?

Reply:
Yes exactly.